src/WebJobs.Script/Extensions/FileUtility.cs

```diff
@@ -66,7 +66,7 @@ public static Task<bool> DeleteIfExistsAsync(string path)
             }
 
             encoding = encoding ?? Encoding.UTF8;
-            using (Stream fileStream = OpenFile(path, FileMode.Create, FileAccess.Write, FileShare.ReadWrite | FileShare.Delete))
+            using (Stream fileStream = OpenFile(path, FileMode.Create, FileAccess.Write, FileShare.Read))
```

Comment:
This used to be the default behavior of this API when it was originally introduced.